generators/ci-cd/templates/_Jenkinsfile

```diff
@@ -156,6 +156,27 @@ node {
 <%_ if (jenkinsIntegrations.includes('docker')) { _%>
 <%= gitLabIndent %>    }
 <%_ } _%>
+<%_ if (jenkinsIntegrations.includes('publishDocker')) { _%>
+
+<%= gitLabIndent %>    def dockerImage
+<%= gitLabIndent %>    stage('build docker') {
+    <%_ if (buildTool === 'maven') { _%>
+<%= gitLabIndent %>        sh "cp -R src/main/docker target/"
```

Comment:
Now it's true that since Jenkins is built in Java, we could use maven/gradle to build the docker image.
As for the publish phase, it seems easier to use the docker pipeline plugin but I guess that could be doable with maven/gradle (with more work) if we want to avoid the dependency on the plugin.